docs/style.md

```diff
@@ -75,3 +75,22 @@ from armory.logs import log
 ```
 
 Exceptions are allowed for import error handling, required import ordering, or in-class/function imports.
+
+## Additional Import Block for Downloaded GitHub Repos
+
+A fourth import block may be added for external package imports that require downloading an external github repo via armory.
+This is typically only used for some baseline models and art experimental attacks.
+This must use the `armory.errors.ExternalRepoImport` context manager as follows (one `with` statement per external repo):
+```
+from armory.errors import ExternalRepoImport
+
+with ExternalRepoImport(
+    repo="colour-science/colour@v0.3.16",
+    experiment="carla_obj_det_dpatch_undefended.json",
+):
+    import colour
+```
+
+`repo` refers to the GitHub repo name (optionally with `@tag`).
+`experiment` refers to the `.json` file in the `scenario_config` directory that uses this repo.
+These repos are specifically *NOT* installed in the armory-supported docker containers and conda environments, and are downloaded at runtime.
```

Comment:
A sentence of "why these are not downloaded" might be helpful.

Reply:
Added